### DISCUSSION TITLE
How to update the data for all the users on a POST request

### DISCUSSION THREAD

**Discussion by faisalsayed10** [vercel/swr#1043] [Q&A] [answered]
Hi,
Suppose there's a user A and B.
I have a database which contains a list of `posts`. All these posts are displayed on the homepage of my website.
Suppose, user B makes a POST request to my database and creates a new `post`.   
Eg:
```js
await axios.post("/api/post", data)
```

So now I want the data (which is displayed on the homepage) to be updated for user B as well as user A using swr. Now the thing is, I don't want to revalidate the data on focus or interval. If user A is looking at the browser window, the data for them should update right away in front of his eyes and should not require him to switch tabs or anything. How can I implement this? I'm searching this for hours and haven't found any answer. This is basically optimistic UI but for each and every user? 

I have looked up on many examples but according to me, its only updating the data for the user who is actually making the POST request and not for other users.

**Comment by ErnestBrandi** [comment]
> Now the thing is, I don't want to revalidate the data on focus or interval.

Why that ? How would you proceed to do what you want with any other tool / library anyway ?

**Comment by faisalsayed10** [comment, marked as answer]
Hmm... This can be done with sockets only i think. I was wondering if it was possible the same way with swr

**Reply by ErnestBrandi** [reply]
If you need something instant then yes use websockets (if available, depending on the framework you use).
Otherwise, `swr` is good enough as a _ui-refresher_.
